pkg/controller/storagecluster/initialization_reconciler.go

```diff
@@ -16,8 +17,23 @@ import (
 	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
 )
 
+type ExternalResources []struct {
+	Kind           string            `json:"kind"`
+	Data           map[string]string `json:"data,omitempty"`
+	Name           string            `json:"name"`
+	PoolName       string            `json:"pool-name,omitempty"`
+	FilesystemName string            `json:"filesystem-name,omitempty"`
+}
+
+const (
+	secretName      = "rook-ceph-external-cluster-details"
+	secretNamespace = "openshift-storage"
```

Comment:
This feels off to have it hardcoded, the op should know from the `ReconcileStorageCluster` I assume.

Reply:
If we won't hardcode it, how we are we going to set the values for `Name` and `Namespace`  for secrets and configmap at line `116` down below?.

Reply:
I don't know, doesn't `ReconcileStorageCluster` provide it to you?